text/trace/0189-instrumentation-layers-suppression.md

```diff
@@ -0,0 +1,222 @@
+# Instrumentation Layers and Suppression
+
+This document describes approach for tracing instrumentation layers and suppressing duplicate layers.
+
+## Motivation
+
+- Provide clarity for instrumentation layers: e.g. DB calls on top of REST API
+- Give mechanism to suppress instrumentation layers for the same convention: e.g. multiple instrumented HTTP clients using each other.
+
+## Explanation
+
+### Spec changes proposal
+
+- Tracing Semantic Conventions: Each span MUST follow a single (besides general and potentially composite), convention, specific to the call it describes. It MUST specify it when creating a span.

```

Comment:
As part of this, we'd also need to define which semantic conventions are composable and which are not.

Reply:
Some semantic conventions define it (e.g. [Couch DB](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md) wants to have HTTP attributes on DB spans). Unless it's already explicitly defined, we say conventions are not composable. 
It looks like we should allow multiple types on the API surface if we want to allow that.